Examples/FullStackSwiftExample/iOSApp/Kittens/ViewModels/AddKittenViewModel.swift

```diff
@@ -0,0 +1,25 @@
+import Models
+import SwiftUI
+
+/// Models the data used in the `AddKitten` view.
+class AddKittenViewModel: ObservableObject {
+    /// New kitten name (initially, an empty string).
+    @Published var name = ""
+    /// New kitten color (initially, an empty string).
+    @Published var color = ""
+    /// New kitten favorite food. We have to display some initial value, so use chicken to start.
+    @Published var favoriteFood: CatFood = .chicken
+
+    /// Sends a request to add a new kitten to the backend.
+    private func addKitten() async throws {
+        let kitten = Kitten(name: name, color: color, favoriteFood: favoriteFood)
+        try await HTTP.post(url: HTTP.baseURL, body: kitten)
+    }
+
+    /// Adds a new kitten and then runs the provided completion handler.
+    func addAction(completion: @escaping () -> Void) {
+        runInTask(completion: completion) {
+            try await self.addKitten()
+        }
+    }
```

Comment:
I'd handle the async stuff in the Views
```suggestion
    /// Sends a request to add a new kitten to the backend.
   func addKitten() async throws {
        let kitten = Kitten(name: name, color: color, favoriteFood: favoriteFood)
        try await HTTP.post(url: HTTP.baseURL, body: kitten)
    }
```